lib/Wikia/src/Service/User/Permissions/PermissionsServiceImpl.php

```diff
@@ -88,6 +89,27 @@ public function getAutomaticGroups( \User $user, $recacheAutomaticGroups = false
 	}
 
 	/**
+	 * Returns list of access control groups that need to be added based on the explicit user groups
```

Comment:
I believe it would be helpful to say not what the code is doing but why is it doing it. I.e. we need to make sure no highly privileged accounts are able to remove restricted-login group which controls 2FA.